### PR TITLE
Update doc

### PR DESCRIPTION
**Pull request by fanny-jourdan**
## Description

- use of mkdocs functionalities on attribution method docs (with details added to docstring in all methods)
- rewrote Attribution method doc overview.md
- fixed a bug in attribution method tests (wrong parameter call for generation)
- minor change to attribution file base.py to add use_gradient as parameter (for greater fluidity in doc writing)

## Type of Change

- 📚 Examples / docs / tutorials / dependencies update
- 🔧 Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I've read the [`CODE_OF_CONDUCT.md`](https://github.com/FOR-sight-ai/interpreto/blob/main/CODE_OF_CONDUCT.md) document.
- [x] I've read the [`CONTRIBUTING.md`](https://github.com/FOR-sight-ai/interpreto/blob/main/docs/contributing.md) guide.
- [x] I've successfully run the style checks using `make lint`.
- [x] I've written tests for all new methods and classes that I created and successfully ran `make test`.
- [x] I've written the docstring in Google format for all the methods and classes that I used.

### CODE REVIEW

File: docs/api/attributions/overview.md [hunk 1]
```diff
@@ -3,68 +3,84 @@
 ## Common API
 
 ```
-explainer = Methods(model, tokenizer, device, args)
-explanation = explainer.explain(inputs, targets)
+from interpreto import Method
+
+explainer = Method(model, tokenizer, device, args)
```
Comment (AntoninPoche) [inline docs/api/attributions/overview.md]: The `device` is not the third argument; the `batch_size` is. I would remove the `, device, args` part, but you can choose what you want to leave.

File: docs/api/attributions/overview.md [hunk 2]
```diff
@@ -3,68 +3,84 @@
...
+**Step 1:** Explainer instantiation: `Method` is a generic name for an attribution method from the list of available methods (below). It inherits from the `AttributionExplainer` base class. Its initialization takes the general parameters explained here (and parameters that are specific to each method, explained in their own documentation):
 
-- explain method: The call to explainer generates the explanations, it takes two parameters:
-  - inputs: the samples on which the explanations are requested, see inputs section for more detail.
-  - targets: another parameter to specify what to explain in the inputs, can be a specific class or a set of classes (for classification), or texts (for generation). If targets=None, the target is calculated by making the prediction from the input with the given model.
+- `model` (`PreTrainedModel`): Hugging Face model to explain,
+- `tokenizer` (`PreTrainedTokenizer`): Hugging Face tokenizer associated with the model,
+- `batch_size` (`int`): batch size for the attribution method,
+- `device` (`torch.device | None = None`): device on which the attribution method will be run,
+- `args`: specifics for each method.
```
Comment (AntoninPoche) [inline docs/api/attributions/overview.md:20]: It might be interesting to talk about granularity and inference modes here. I know they do not apply to all methods, but users might want to know about them.

File: docs/api/attributions/overview.md [hunk 3]
```diff
@@ -3,68 +3,84 @@
...
 **Gradient based methods:**
 
-- Saliency, InputxGradient: [Simonyan et al. 2013, Deep Inside Convolutional Networks: Visualising Image Classification Models and Saliency Maps](https://arxiv.org/abs/1312.6034).
-- Integrated Gradient: [Sundararajan et al. 2017, Axiomatic Attribution for Deep Networks](http://proceedings.mlr.press/v70/sundararajan17a.html).
-- SmoothGrad: [Smilkov et al. 2017, SmoothGrad: removing noise by adding noise](https://arxiv.org/abs/1706.03825)
+- [Integrated Gradient](./methods/integrated_gradients.md): [Sundararajan et al. 2017, Axiomatic Attribution for Deep Networks](http://proceedings.mlr.press/v70/sundararajan17a.html).
+- [Saliency](./methods/saliency.md): [Simonyan et al. 2013, Deep Inside Convolutional Networks: Visualising Image Classification Models and Saliency Maps](https://arxiv.org/abs/1312.6034).
+- [SmoothGrad](./methods/smoothgrad.md): [Smilkov et al. 2017, SmoothGrad: removing noise by adding noise](https://arxiv.org/abs/1706.03825)
 
 # Custom API
```
Comment (AntoninPoche) [inline docs/api/attributions/overview.md:47]: I do not think this part is clear. I would keep it simple:

You can easily use Interpreto with your attribution methods. You have to implement two things:

- a `Perturbator` to generate perturbed elements.
- an `Aggregator` to aggregate scores on perturbed elements.

Then, inherit from `MultitaskExplainerMixin` and `AttributionExplainer`, and finally, in the initialization of your method, point to your perturbator and aggregator in the `super().__init__`.

Please check how we did it with the implemented method. If you have any questions, please file an issue or contact us directly.

Once you succeed, please make a pull request. We welcome your method and contributions to the library.

File: interpreto/attributions/methods/sobol_attribution.py [hunk 4]
```diff
@@ -45,13 +45,27 @@
...
+    Fel et al. (2021). *Look at the variance! Efficient black-box explanations with Sobol-based sensitivity analysis.*
+    [Paper](https://arxiv.org/abs/2111.04138)
+
+    Examples:
+        >>> from interpreto import Granularity, Sobol
+        >>> from interpreto.attributions import InferenceModes
+        >>> method = Sobol(model, tokenizer, batch_size=4,
+        >>>                inference_mode=InferenceModes.LOGITS,
+        >>>                n_token_perturbations=8,
+        >>>                granularity=Granularity.WORD)
```
Comment (AntoninPoche) [inline interpreto/attributions/methods/sobol_attribution.py]: You can add two arguments here so that people know how they should be used:

```
        >>>                sobol_indices_order=Sobol.sobol_indices_orders.FIRST_ORDER,
        >>>                sampler=Sobol.samplers.SOBOL)
```

File: mkdocs.yml [hunk 5]
```diff
@@ -32,13 +32,16 @@ nav:
         - Sparsity: api/concepts/metrics/sparsity_metrics.md
     - Attribution Explainers:
       - Overview: api/attributions/overview.md
-      - Methods: 
+      - Methods:
+        - KernelSHAP: api/attributions/methods/kernelshap.md
+        - LIME: api/attributions/methods/lime.md
         - Integrated Gradients: api/attributions/methods/integrated_gradients.md
```
Comment (AntoninPoche) [inline mkdocs.yml]: I think it is better if the appearing name corresponds to the class name.

`KernelShap`
`IntegratedGradients`

File: mkdocs.yml [hunk 6]
```diff
@@ -32,13 +32,16 @@ nav:
...
         - Occlusion: api/attributions/methods/occlusion.md
         - Saliency: api/attributions/methods/saliency.md
         - SmoothGrad: api/attributions/methods/smoothgrad.md
-    - Evaluation Metrics:
-      - Overview: api/evaluation_metrics/overview.md
+        - Sobol: api/attributions/methods/sobol.md
+      - Metrics:
+        - Overview: api/attributions/metrics/overview.md
```
Comment (AntoninPoche) [inline mkdocs.yml]: For now, when we click on `API Reference`, it opens `Model Wrapping: api/commons/model_wrapping.md`.

If you move the `Attribution Explainers` part before the line above, it would appear first.

File: mkdocs.yml
Comment (AntoninPoche) [inline mkdocs.yml:1]: I know you did not touch these parts, but several pages are empty.

You can remove `Quickstart` and `Roadmap` (or fill them in).

Regarding `Tutorials,` it would be best to show our two notebooks here. But this may require more work, like adding another package to the library.

File: docs/api/attributions/overview.md [hunk 7]
```diff
@@ -3,68 +3,84 @@
...
 
-TODO: add link to documentation of each method. A table would be easier to read.
+
+**Step 2:** The `AttributionExplainer` class overloads the `__call__` method to directly invoke the `explain` function. Therefore, calling `explainer(inputs, targets)` is equivalent to `explainer.explain(inputs, targets)`. It takes two parameters:
+
+- `inputs`: the samples for which explanations are requested.
+- `targets`: specifies what to explain in the inputs. This can be a specific class or a set of classes (for classification tasks), or target texts (for generation tasks). If `targets=None`, the target is automatically inferred by performing a prediction on the input using the provided model.
+
+
+## Specifics methods
```
Comment (AntoninPoche) [inline docs/api/attributions/overview.md]: The title is weird. Maybe `Available methods`?

**Review by AntoninPoche — CHANGES_REQUESTED** [review]
Great work @fanny-jourdan  !!!

The documentation is great! I left some minor comments.

Finally, I have a suggestion. I do not know if you saw in the concept-based documentation, but in front of the page names, there is a small icon. Like arrows and such. I think it would be great for attribution methods to.

To add them, the following lines should be added at the beginning of the `your_method.md`:

```
---
icon: material/name_of_your_icon
---
```

You can find the names here: https://pictogrammers.com/library/mdi/ . There are others, ask me if you do not find something that pleases you.

I suggest using:
- `arrow-expand-right` for perturbation-based methods
- `swap-horizontal` for gradient-based methods

File: requirements.txt [hunk 8]
```diff
@@ -79,6 +79,7 @@ matplotlib==3.10.3
     # via overcomplete
 matplotlib-inline==0.1.7
     # via ipython
+mknotebooks
```
Comment (AntoninPoche) [inline requirements.txt]: The requirements should not be added to the `requirements.txt` but to the `pyproject.toml` which automatically creates the two requirements files through `make update-deps`.

I will modify it, do not worry.

Comment (fanny-jourdan) [reply requirements.txt]: thanks ! :)
